src/panels/calendar/ha-panel-calendar.ts

```diff
@@ -0,0 +1,304 @@
+
+const palette = [
+  "ff0029",
+  "66a61e",
+  "377eb8",
+  "984ea3",
+  "00d2d5",
+  "ff7f00",
+  "af8d00",
+  "7f80cd",
+  "b3e900",
+  "c42e60",
+  "a65628",
+  "f781bf",
+  "8dd3c7",
+  "bebada",
+  "fb8072",
+  "80b1d3",
+  "fdb462",
+  "fccde5",
+  "bc80bd",
+  "ffed6f",
+  "c4eaff",
+  "cf8c00",
+  "1b9e77",
+  "d95f02",
+  "e7298a",
+  "e6ab02",
+  "a6761d",
+  "0097ff",
+  "00d067",
+  "f43600",
+  "4ba93b",
+  "5779bb",
+  "927acc",
+  "97ee3f",
+  "bf3947",
+  "9f5b00",
+  "f48758",
+  "8caed6",
+  "f2b94f",
+  "eff26e",
+  "e43872",
+  "d9b100",
+  "9d7a00",
+  "698cff",
+  "d9d9d9",
+  "00d27e",
+  "d06800",
+  "009f82",
+  "c49200",
+  "cbe8ff",
+  "fecddf",
+  "c27eb6",
+  "8cd2ce",
+  "c4b8d9",
+  "f883b0",
+  "a49100",
+  "f48800",
+  "27d0df",
+  "a04a9b",
+];
+
+@customElement("ha-panel-calendar")
+class PanelCalendar extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true, attribute: "narrow" })
+  public narrow = false;
+
+  @property() private _calendars: SelectedCalendar[] = [];
+
+  @property() private _events: CalendarEvent[] = [];
+
+  private _start?: Date;
+
+  private _end?: Date;
+
+  private _firstUpdate = false;
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    if (!this.hass) {
+      return;
+    }
+
+    this._fetchCalendars();
+  }
+
+  protected updated() {
+    if (!this._firstUpdate) {
+      this._fetchData();
+    }
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <app-header-layout has-scrolling-region>
+        <app-header fixed slot="header"
+          ><app-toolbar>
+            <ha-menu-button
+              .hass=${this.hass}
+              .narrow=${this.narrow}
+            ></ha-menu-button>
+            <div main-title>${this.hass.localize("panel.calendar")}</div>
+          </app-toolbar>
+        </app-header>
+        <div class="content">
+          <div class="calendar-list">
+            <div class="calendar-list-header">My Calendars</div>
+            ${this._calendars.map(
+              (selCal) =>
+                html`<div>
+                  <mwc-formfield .label=${selCal.calendar.name}>
+                    <mwc-checkbox
+                      style=${styleMap({
+                        "--mdc-theme-secondary": selCal.backgroundColor,
+                      })}
+                      .value=${selCal.calendar.entity_id}
+                      .checked=${selCal.selected}
+                      @change=${this._handleToggle}
+                    ></mwc-checkbox>
+                  </mwc-formfield>
+                </div>`
+            )}
+          </div>
+          <ha-full-calendar
+            .events=${this._events}
+            @view-changed=${this._handleViewChanged}
+          ></ha-full-calendar>
+        </div>
+      </app-header-layout>
+    `;
+  }
+
+  private _fetchCalendars() {
+    this.hass.callApi<Calendar[]>("GET", "calendars").then((result) => {
```

Comment:
The calendars are entities that are in `hass.states` right? So no need to do an API request? We should also check for updates of the state to refetch the events.

Reply:
Will look into this tomorrow